packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1339,8 +1359,9 @@ export class GarbageCollector implements IGarbageCollector {
      * @param gcData - The data representing the reference graph on which GC is run.
      * @param gcResult - The result of the GC run on the gcData.
      * @param currentReferenceTimestampMs - The timestamp to be used for unreferenced nodes' timestamp.
+     * @returns - A list of nodes that are ready to be deleted
      */
-    private updateCurrentState(
+    private updateCurrentMarkState(
```

Comment:
What do you think of these names:
- Rename this function `runMarkPhase` since it basically marks nodes as unreferened and adds unreference timestamp.
- Rename `sweepNodes` to `runSweepPhase` since it deletes nodes.
- Add a new function `updateCurrentState` that is run after sweep. It calls this.newReferencesSinceLastRun.clear, updates this.gcDataFromLastRun along with removal of deleted nodes and updates this.deletedNodes.

Reply:
Also, this.tombstones = [] doesn't need to be done where since it is done when calling runtime.updateTombstoneRoutes.

Reply:
Yeah, this is a great idea